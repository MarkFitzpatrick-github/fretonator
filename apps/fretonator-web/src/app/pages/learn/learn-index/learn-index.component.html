<app-stripe-banner></app-stripe-banner>

<div class="learn__container">
  <h1 class="learn__title">Learn Guitar Theory</h1>
  <p class="learn__introCopy">Welcome to the Fretonator learning hub, where you can learn music theory tailored to the
    guitar, guitar techniques and essential maintenance skills.</p>
  <p class="learn__introCopy">We'll be adding new content regularly, so please keep checking back!</p>

  <hr class="learn__divider"/>

  <a [routerLink]="['/', 'learn', 'patterns']"
     [state]="{scrollToTop: true}"
     class="learn__item"
     aria-label="Learn patterns">
    <div class="learn__imageContainer">
      <img src="/assets/img/3nps.png" class="learn__image" alt="Patterns"/>
      <a [routerLink]="['/', 'learn', 'patterns']"
         [state]="{scrollToTop: true}"
         class="learn__link">Learn patterns</a>
    </div>
    <div class="learn__info">
      <h2 class="learn__itemTitle">Patterns</h2>
      <p class="learn__itemExcerpt">There are only 7 patterns you need to learn on the guitar. Once you've memorised
        those patterns, you can play any of the 7 standard modes from any starting note, anywhere on the fretboard.</p>
    </div>
  </a>

</div>
