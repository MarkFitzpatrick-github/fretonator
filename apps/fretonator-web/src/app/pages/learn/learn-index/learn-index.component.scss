@import '../../../../styles/vars';
@import '../../../../styles/mixins';
@import '../../../../styles/typography';

.learn__container {
  @include content_wrapper();
  max-width: $screen-med;
}

.learn__title {
  @include font_large();
  text-align: center;
  margin-bottom: pxToRem($grid-unit * 4);
  margin-top: pxToRem($grid-unit * 4);
}

.learn__introCopy {
  @include font_bodyCopy();
  margin-bottom: pxToRem($grid-unit * 4);
}

.learn__divider {
  @include divider();
}

.learn__item {
  display: flex;
  flex-direction: column;
  margin-top: pxToRem($grid-unit * 6);
  margin-bottom: pxToRem($grid-unit * 6);
  border: pxToRem(2) solid var(--black);
  color: var(--foreground-color-base);
  text-decoration: none;
  box-shadow: 8px 8px 0 0 var(--grey);

  @media screen and (min-width: $screen-med) {
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-start;
  }
}

.learn__imageContainer {
  padding: pxToRem($grid-unit * 4);
  background-color: var(--black);
  margin-bottom: 0;
  display: flex;
  flex-direction: column;

  @media screen and (min-width: $screen-sm) {
    flex-basis: 42%;
    flex-shrink: 0;
  }
}

.learn__image {
  width: 100%;
  border: pxToRem(2) solid var(--yellow);
  box-shadow: 8px 8px 0 0 var(--yellow);
}

.learn__info {
  display: flex;
  flex-direction: column;
  padding: pxToRem($grid-unit * 4);
}

.learn__itemTitle {
  @include font_medium();
  margin-bottom: pxToRem($grid-unit * 3);
  color: var(--foreground-color-base);
  text-decoration: none;
}

.learn__itemExcerpt {
  @include font_bodyCopy();
  margin-bottom: pxToRem($grid-unit * 4);
  color: var(--foreground-color-base);
  text-decoration: none;
}

.learn__link {
  @include hard_button_base();
  margin-bottom: pxToRem($grid-unit);
}
