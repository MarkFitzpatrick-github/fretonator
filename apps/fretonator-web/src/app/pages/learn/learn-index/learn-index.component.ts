import { Component, OnInit } from '@angular/core';
import { MetaService } from '../../../common/meta/meta.service';

@Component({
  selector: 'app-learn-index',
  templateUrl: './learn-index.component.html',
  styleUrls: ['./learn-index.component.scss']
})
export class LearnIndexComponent implements OnInit {
  pageDescription = 'Welcome to the Fretonator learning hub, where you can learn music theory tailored to the guitar, guitar techniques and essential maintenance skills.';
  pageTitle = 'Learn guitar theory with the Fretonator - the ultimate interactive free guitar theory tool';
  pageUrl = 'https://www.fretonator.com/learn';

  constructor(private metaService: MetaService) {
  }

  ngOnInit(): void {
    this.metaService.updateAllGenericMeta(this.pageUrl, this.pageTitle, this.pageDescription);
  }
}
