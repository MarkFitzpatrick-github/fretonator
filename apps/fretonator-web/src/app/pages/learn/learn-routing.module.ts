import { NgModule } from '@angular/core';
import { Route, RouterModule, Routes } from '@angular/router';
import { LearnIndexComponent } from './learn-index/learn-index.component';
import { PatternsIndexComponent } from './patterns-index/patterns-index.component';
import { ScrollToGuard } from '../../common/scroll-to/scroll-to.guard';

const learnHub: Route = {
  path: '',
  pathMatch: 'full',
  canActivate: [
    ScrollToGuard
  ],
  component: LearnIndexComponent,
};

const patterns: Route = {
  path: 'patterns',
  pathMatch: 'full',
  canActivate: [
    ScrollToGuard
  ],
  component: PatternsIndexComponent,
};

const routes: Routes = [learnHub, patterns];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule]
})
export class LearnRoutingModule { }
