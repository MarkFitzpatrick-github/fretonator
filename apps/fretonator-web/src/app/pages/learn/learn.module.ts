import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { LearnRoutingModule } from './learn-routing.module';
import { LearnIndexComponent } from './learn-index/learn-index.component';
import { StripeBannerModule } from '../../common/stripe-banner/stripe-banner.module';
import { PatternsIndexComponent } from './patterns-index/patterns-index.component';
import { FretboardModule } from '../../common/fretonator/fretboard/fretboard.module';


@NgModule({
  declarations: [
    LearnIndexComponent,
    PatternsIndexComponent
  ],
  imports: [
    CommonModule,
    LearnRoutingModule,
    StripeBannerModule,
    FretboardModule
  ],
  exports: [
    LearnIndexComponent,
    PatternsIndexComponent
  ]
})
export class LearnModule {
}
