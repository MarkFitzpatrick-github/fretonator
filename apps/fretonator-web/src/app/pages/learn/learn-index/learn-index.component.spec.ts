import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { LearnIndexComponent } from './learn-index.component';
import { Component } from '@angular/core';
import { LearnModule } from '../learn.module';
import { RouterTestingModule } from '@angular/router/testing';

describe('LearnIndexComponent', () => {
  @Component({
    selector: 'app-learn-index-spec',
    template: `
      <app-learn-index></app-learn-index>
    `
  })
  class LearnIndexComponentSpec {
  }
  let component: LearnIndexComponentSpec;
  let fixture: ComponentFixture<LearnIndexComponentSpec>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ LearnIndexComponentSpec ],
      imports: [
        LearnModule,
        RouterTestingModule
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(LearnIndexComponentSpec);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
