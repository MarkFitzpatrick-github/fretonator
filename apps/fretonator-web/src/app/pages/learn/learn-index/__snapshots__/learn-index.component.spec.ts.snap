// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnIndexComponent should create 1`] = `LearnIndexComponentSpec {}`;
